.drone.yml

```diff
@@ -42,43 +43,38 @@ pipeline:
 
   # pr
 
-  sbt-update:
-    <<: *sbtenv
-    commands:
-      - sbt update
-
   compile:
     <<: *sbtenv
     commands:
-      - sbt rholang/bnfc:generate compile test:compile it:compile doc
-
-  run-unit-tests:
-    <<: *sbtenv
-    commands:
-      - sbt test it:test
-
-  run-code-analysis:
-    <<: *sbtenv
-    commands:
-      - sbt scalafmtCheck coverage
-      - ./scripts/ci/codecov-upload
+      - sbt update rholang/bnfc:generate compile
 
   # push
 
-  package:
+  docker-build:
     <<: *sbtenv
     commands:
-      - sbt node/docker:publishLocal node/debian:packageBin node/universal:packageZipTarball
-      - mkdir -p artifacts
-      - cp node/target/rnode_*_all.deb node/target/universal/*.tgz artifacts/
+      # Skip the `doc` task - it's being invoked explicitly later
+      - SKIP_DOC=true sbt node/docker:publishLocal
     volumes:
       - /var/cache/rchain-build/.sbt:/root/.sbt
       - /var/cache/rchain-build/.ivy2:/root/.ivy2
       - /var/run/docker.sock:/var/run/docker.sock
     when:
       event: [ push, tag ]
 
+  # pr
+
+  run-unit-tests:
+    group: parallel
+    <<: *sbtenv
+    commands:
+      - cp -r . ../step.run-unit-tests && cd ../step.run-unit-tests
+      - sbt test it:test
```

Comment:
sbt it:test runs unit tests?

Reply:
I'm not sure. Does it? I'd leave it as is so that nobody has to wonder about that :)

Reply:
I'd say this should run together with integration tests. Doesn't need to change within this PR though

Reply:
I'm not even sure what `it:test` is :sweat_smile:

Reply:
I gave it a run 

```
$ sbt it:test
[info] InMemoryStoreConcurrencyTests:
[info] CORE-589 produce and consume in a multithreaded fashion
Finished InMemory: 500 iterations.
Time per thread: avg: 7309.80 ms; min: 7292.59 ms; max: 7332.91 ms
[info] - should show high performance (11 seconds, 84 milliseconds)
[info] produce and consume with monix.Tasks
Finished InMemory: 500 iterations.
Time per thread: avg: 955.43 ms; min: 780.50 ms; max: 1207.45 ms
[info] - should work (5 seconds, 679 milliseconds)
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by org.lmdbjava.ByteBufferProxy$AbstractByteBufferProxy (file:/Users/kent/.ivy2/cache/org.lmdbjava/lmdbjava/jars/lmdbjava-0.6.1.jar) to field java.nio.Buffer.address
WARNING: Please consider reporting this to the maintainers of org.lmdbjava.ByteBufferProxy$AbstractByteBufferProxy
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
[info] LMDBStoreConcurrencyTestsWithTls:
[info] CORE-589 produce and consume in a multithreaded fashion
Finished LMDB with TLS: 500 iterations.
Time per thread: avg: 25448.36 ms; min: 25345.17 ms; max: 25493.15 ms
[info] - should show high performance (28 seconds, 316 milliseconds)
[info] produce and consume with monix.Tasks
Finished LMDB with TLS: 500 iterations.
Time per thread: avg: 3695.90 ms; min: 3139.30 ms; max: 4172.58 ms
[info] - should work (19 seconds, 963 milliseconds)
[info] LMDBStoreConcurrencyTestsNoTls:
[info] CORE-589 produce and consume in a multithreaded fashion
Finished LMDB NO_TLS: 500 iterations.
Time per thread: avg: 18583.58 ms; min: 18559.59 ms; max: 18604.44 ms
[info] - should show high performance (18 seconds, 806 milliseconds)
[info] produce and consume with monix.Tasks
Finished LMDB NO_TLS: 500 iterations.
Time per thread: avg: 4004.09 ms; min: 3999.59 ms; max: 4008.89 ms
[info] - should work (20 seconds, 342 milliseconds)
[info] ScalaTest
[info] Run completed in 1 minute, 56 seconds.
[info] Total number of tests run: 6
[info] Suites: completed 3, aborted 0
[info] Tests: succeeded 6, failed 0, canceled 0, ignored 0, pending 0
[info] All tests passed.
[info] Passed: Total 6, Failed 0, Errors 0, Passed 6
[success] Total time: 156 s, completed Dec 14, 2018, 1:15:10 AM
```

Reply:
I see; yeah I agree with @lukasz-golebiewski that this should go together with integration tests in the future

Reply:
So it doesn't run unit tests. Having this moved to the integration-test step might allow squeezing another 2-3 mins due to better parallelization. Let's keep it for latter PRs though :)